apps/dcos_dns/include/dcos_dns.hrl

```diff
@@ -7,6 +7,9 @@
 -define(SPIRAL, spiral).
 
 -type upstream() :: {inet:ip4_address(), inet:port_number()}.
+-type zone() :: binary().
+-type forward() :: {[upstream()] | internal, zone()}.
+-type forward_status() :: unresolved | resolved | failed | timeout.
```

Comment:
There is only one function (i.e. `resolve_loop/n`) that uses this type. It should be defined in its type spec.